packages/shared/src/store/sync.ts

```diff
@@ -17,14 +17,23 @@ const logger = createDevLogger('sync', false);
 
 export const syncInitData = async () => {
   const initData = await syncQueue.add('init', () => api.getInitData());
+  const writers = initData.channelPerms.flatMap((chanInit) =>
```

Comment:
There are a number of places where we check array length inside query write methods -- could this go there too, for consistency?

Reply:
Yeah. It looked like I couldn't early return within `insertChannelPerms` because the type def wouldn't accept a return type of undefined, but after marking the queryFn as async typescript doesn't seem to mind (kinda weird), pushing fix.